core/controllers/base_test.py

```diff
@@ -831,8 +831,8 @@ def test_responses_with_valid_iframe_restriction(self):
         self.get_html_response('/mock')
 
         response = self.get_html_response(
-            '/mock', params={'iframe_restriction': 'DENY'})
-        self.assertEqual(response.headers['X-Frame-Options'], 'DENY')
+            '/mock', params={str('iframe_restriction'): str('DENY')})
+        self.assertEqual(response.headers[str('X-Frame-Options')], str('DENY'))
 
         response = self.get_html_response(
             '/mock', params={'iframe_restriction': 'SAMEORIGIN'})
```

Comment:
Why do these lines work then (that is, without the conversion)?

Reply:
Hi @apb7,

i also thought about it but seeing the tests failing [here](https://circleci.com/gh/oppia/oppia/9339?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) it clearly mentioned that 'u'DENY' is not a valid latin1 string' that's why I only str() it and tested again 

<img width="1106" alt="Screenshot 2019-06-18 at 1 31 49 AM" src="https://user-images.githubusercontent.com/41017303/59641133-0e5bc680-9169-11e9-9803-43bab2445dcc.png">

Reply:
I don't think that this is valid reasoning. I agree that only a particular test fails but we should be clear what's happening behind the scenes and why the other test does not fail.

Reply:
Hi @apb7, I've updated the changes with reason PTAL Thanks!